docs/design-docs/multi-DM.md

```diff
@@ -66,3 +70,57 @@ behaviors, namely:
      LockingTransactions.client - to ensure that the lock identifier
      is consistent across all replicas.  This change should be
      straightforward.
+     
+## AtLeastOnceRPC + DHT + ConsensusRSM
+
+### Desired behavior
+
+1. Client creates a new instance or obtains a reference to an existing MicroService.
+2. Client makes a call to an application method in Microservice and gets the result back.
+3. Client DM retries on failure for certain duration.
+4. Microservice replicas are sharded.
+5. Client expects the MicroService to be highly available,
+   resilient to server machine failures (provided that concurrent
+   failures are limited to a minority quorum).
+6. Client expects the MicroService to be high performance (all
+   quorum communication is on the local zone network).
+7. Client does not expect the Microservice to be resilient to zone failure.   
+   
+### How it works under the hood
+
+1. Client creates an instance of a MicroService (create()).
+	1. Kernel invokes group.onCreate() on all DM's in the order of inner most DM to outer most DM.
+		1. AtLeastOnceRPCPolicy.GroupPolicy.onCreate() does nothing unusual.	
+		1. DHTPolicy.GroupPolicy.onCreate() ensures that each Microservice is distributed
```

Comment:
This is not true.  We're still busy creating this instance here.

Reply:
Perhaps you mean that it creates n shards, and configures the key range for each shard?  It's somewhat vague in this text whether at this point each shard has a single replica.  I assume so, and that that replica is passed to ConsensusRSMPolicy.GroupPolicy.onCreate() which creates the additional 2f replicas.

Reply:
Changed the order of calling group.onCreate() from Consensus > DHT > AtLeast.
Added the following details:
1. DHTPolicy.GroupPolicy.onCreate() creates n shards, and configures the key range for each shard.
			1. Server creates an instance of DMs for each shard.
			1. When creating each shard, group.onCreate() is invoked in the order of inner most DM to the last DM before the current DM (ConsensusRSMPolicy.GroupPolicy).
			1. ConsensusRSMPolicy.GroupPolicy creates 2f+1 replicas (by invoking 
		replicate(), which in turn invokes addServer on all DM's).